lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1471,7 +1471,13 @@ static LogicalResult canonicalizeSingleSetConnect(StrictConnectOp op,
     }
   }
 
-  if (isUselessName(connectedDecl)) {
+  // TODO: This is a hack to prevent removal if the operation is used by the
+  // Grand Central pass.  This should be changed later to use something less
+  // brittle.
+  auto gctTap =
+      AnnotationSet(connectedDecl)
+          .hasAnnotation("sifive.enterprise.grandcentral.AugmentedGroundType");
+  if (isUselessName(connectedDecl) && !gctTap) {
```

Comment:
We have ways of keeping things around, we don't need to, nor should we, poke at specific annotations.